src/Catalyst.Core.Lib/Dag/DagLink.cs

```diff
@@ -0,0 +1,176 @@
+    /// </summary>
+    public class DagLink : IMerkleLink
+    {
+        /// <summary>
+        ///   Create a new instance of <see cref="DagLink"/> class.
+        /// </summary>
+        /// <param name="name">The name associated with the linked node.</param>
+        /// <param name="id">The <see cref="Cid"/> of the linked node.</param>
+        /// <param name="size">The serialised size (in bytes) of the linked node.</param>
+        public DagLink(string name, Cid id, long size)
+        {
+            Name = name;
+            Id = id;
+            Size = size;
+        }
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="DagLink"/> class from the
+        ///   specified <see cref="IMerkleLink"/>.
+        /// </summary>
+        /// <param name="link">
+        ///   Some type of a Merkle link.
+        /// </param>
+        public DagLink(IMerkleLink link)
+        {
+            Name = link.Name;
+            Id = link.Id;
+            Size = link.Size;
+        }
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="DagLink"/> class from the
+        ///   specified <see cref="Stream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   A <see cref="Stream"/> containing the binary representation of the
+        ///   <b>DagLink</b>.
+        /// </param>
+        public DagLink(Stream stream) { Read(stream); }
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="DagLink"/> class from the
+        ///   specified <see cref="CodedInputStream"/>.
+        /// </summary>
+        /// <param name="stream">(
+        ///   A <see cref="CodedInputStream"/> containing the binary representation of the
+        ///   <b>DagLink</b>.
+        /// </param>
+        public DagLink(CodedInputStream stream) { Read(stream); }
+
+        /// <inheritdoc />
+        public string Name { get; private set; }
+
+        /// <inheritdoc />
+        public Cid Id { get; private set; }
+
+        /// <inheritdoc />
+        public long Size { get; private set; }
+
+        /// <summary>
+        ///   Writes the binary representation of the link to the specified <see cref="Stream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="Stream"/> to write to.
+        /// </param>
+        public void Write(Stream stream)
+        {
+            using (var cos = new CodedOutputStream(stream, true))
+            {
+                Write(cos);
+            }
+        }
+
+        /// <summary>
+        ///   Writes the binary representation of the link to the specified <see cref="CodedOutputStream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="CodedOutputStream"/> to write to.
+        /// </param>
+        public void Write(CodedOutputStream stream)
+        {
+            if (stream == null)
+                throw new ArgumentNullException(nameof(stream));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name '' is not declared in the argument list.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)